www/include/views/componentTemplates/DB-Func.php

```diff
@@ -67,26 +83,39 @@ function NameHsrTestExistence($name = null)
 
 function DsHsrTestExistence($name = null)
 {
-    global $pearDB, $pearDBO, $form;
-    $gsvs = null;
+    global $pearDB, $form;
+    $formValues = array();
     if (isset($form)) {
-        $gsvs = $form->getSubmitValues();
+        $formValues = $form->getSubmitValues();
+    }
+    $query = "SELECT compo_id FROM giv_components_template WHERE name = '" . CentreonUtils::escapeSecure($name) . "'";
+
+    if (!empty($formValues['host_id'])) {
+        if (preg_match('/([0-9]+)-([0-9]+)/', $formValues['host_id'], $matches)) {
+            $formValues['host_id'] = (int) $matches[1];
+            $formValues['service_id'] = (int) $matches[2];
+        } else {
+            throw new \InvalidArgumentException('host_id must be a combination of integers');
+        }
```

Comment:
Why not using your method `parseHostIdPostParameter` ?

Reply:
Can't be applied for this case